src/libostree/ostree-sysroot.c

```diff
@@ -2243,7 +2243,8 @@ ostree_sysroot_deployment_unlock (OstreeSysroot *self, OstreeDeployment *deploym
                         deployment, _OSTREE_SYSROOT_DEPLOYMENT_RUNSTATE_FLAG_DEVELOPMENT)
                   : _ostree_sysroot_get_runstate_path (
                         deployment, _OSTREE_SYSROOT_DEPLOYMENT_RUNSTATE_FLAG_TRANSIENT);
-        g_autofree char *devpath_parent = dirname (g_strdup (devpath));
+        g_autofree char *devpath_copy = g_strdup (devpath);
+        g_autofree char *devpath_parent = dirname (devpath_copy);
```

Comment:
I don't think that's really a leak like in the previous patch. However, I think it's clearer to just use `g_path_get_dirname` since it acts the way you expect instead of modifying the string in place like `dirname` does.